### PR TITLE
chore(typing): Initial test of typing and introducing a mypy config file

### PR DESCRIPTION
**Pull request by wedamija**
This is a test of converting a file to be entirely typed. I chose something fairly self contained,
and typed and functions imported from other modules as well.

I typed our `json.loads` function as `Any`. I'm open to being more specific here, although from what
I've been reading it's hard to do this well until MyPy supports recursive types.

Also added in a mypy.ini, which enables strict checking and specifies a list of files to run the
checks on. I figure that the way we'd use this is to slowly convert files over time and add them to
this list, and once and entire module is added we'd add a glob match for that module instead.

Looking for feedback on this approach. I think that we can fairly easily have this run in CI using
the config file.

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+warn_return_any=True
+no_implicit_reexport=True
+
+
+[mypy-confluent_kafka.*]
+ignore_missing_imports = True
+[mypy-jsonschema]
+ignore_missing_imports = True
+[mypy-django.*]
+ignore_missing_imports = True
```
Comment (wedamija) [inline mypy.ini:26]: We might be able to get type stubs for some of these, but not trying to just yet.

File: src/sentry/db/models/manager.py [hunk 2]
```diff
@@ -263,7 +264,7 @@ def contribute_to_class(self, model, name):
         super(BaseManager, self).contribute_to_class(model, name)
         class_prepared.connect(self.__class_prepared, sender=model)
 
-    def get_from_cache(self, **kwargs):
+    def get_from_cache(self, **kwargs) -> t.Any:  # TODO(typing): Properly type this
```
Comment (wedamija) [inline src/sentry/db/models/manager.py:267]: We might be able to hook into whatever `django-stubs` is doing to get this to work.

**Review by wedamija — COMMENTED** [review]
We could also potentially use https://github.com/typeddjango/django-stubs to get stubs for our Django models etc, although looks like it requires Django 2.2.x. We need to be on that for Py 3.8 anyway, so maybe that'll happen soon.

Also open to feedback on the `import typing as t` thing. It seems convenient and better than having to import all the specific types you need in every module.

File: src/sentry/db/models/manager.py (hunk 2 above)
Comment (wedamija) [reply src/sentry/db/models/manager.py:267]: Could also at least return that this will be a model

File: src/sentry/db/models/manager.py [hunk 3]
```diff
@@ -263,7 +263,7 @@ def contribute_to_class(self, model, name):
         super(BaseManager, self).contribute_to_class(model, name)
         class_prepared.connect(self.__class_prepared, sender=model)
 
-    def get_from_cache(self, **kwargs):
+    def get_from_cache(self, **kwargs) -> Model:  # TODO(typing): Properly type this
```
Comment (wedamija) [inline src/sentry/db/models/manager.py:266]: We might be able to hook into whatever django-stubs is doing to get this to work.

**Review by armenzg — COMMENTED** [review]
The approach seems to make sense.

Is `mypy` to be added to the dev requirements?

**Comment by armenzg** [comment]
Do you have any instructions on how to activate it for VS Code? or even how to use it?

**Review by untitaker — APPROVED** [review]
I think we can just add mypy to CI, as optional check, and merge this. For VSCode as well as any other IDE pylance/pyright is probably the best.

File: src/sentry/db/models/manager.py (hunk 3 above)
Comment (untitaker) [reply src/sentry/db/models/manager.py:266]: I don't think you can type this without mypy plugins so I would not bother

**Comment by wedamija** [comment]
> Do you have any instructions on how to activate it for VS Code? or even how to use it?

For now:
```
pip install mypy
mypy --config-file mypy.ini
```

I don't have instructions for typing in your ide. But you can use pyright or whatever you like to get good type information. In terms of enforcing typing, mypy seems better.

File: src/sentry/db/models/manager.py (hunk 3 above)
Comment (wedamija) [reply src/sentry/db/models/manager.py:266]: Is there some reason we wouldn't want to use mypy plugins? I think django-stubs is a plugin as well

Comment (untitaker) [reply src/sentry/db/models/manager.py:266]: pyright doesn't support them

**Comment by untitaker** [comment]
fwiw recommended extensions already appear to install pylance for you

**Pull request merged**